src/main/java/org/gatorgradle/task/GatorGradleReportTask.java

```diff
@@ -0,0 +1,123 @@
+package org.gatorgradle.task;
+
+import org.gatorgradle.task.GatorGradleTask;
+import org.gatorgradle.command.BasicCommand;
+import org.gatorgradle.command.Command;
+import org.gatorgradle.config.GatorGradleConfig;
+import org.gatorgradle.display.CommandOutputSummary;
+import org.gatorgradle.internal.DependencyManager;
+import org.gatorgradle.internal.ProgressLoggerWrapper;
+import org.gatorgradle.util.Console;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.workers.WorkerExecutor;
+
+public class GatorGradleReportTask extends GatorGradleTask {
+
+  @Inject
+  public GatorGradleReportTask(WorkerExecutor executor) {
+    this.executor = executor;
+  }
+
+  /**
+   * Execute the grading checks assigned to this GatorGradleReportTask.
+   */
+  @TaskAction
+  public void grade() {
+
+    config.parseHeader();
+
+    // ensure GatorGrader and dependencies are installed
+    String dep = DependencyManager.manage();
+    if (!dep.isEmpty()) {
+      throw new GradleException(dep);
+    }
+
+    Console.newline(1);
+
+    if (config.hasStartupCommand()) {
+      Console.log("Starting up...");
+      BasicCommand startup = (BasicCommand) config.getStartupCommand();
+      startup.outputToSysOut(true);
+      startup.run();
+      if (startup.exitValue() != Command.SUCCESS) {
+        throw new GradleException(
+            "Startup command '" + startup + "' failed with exit code "
+            + startup.exitValue() + "!"
+        );
+      }
+      Console.log("Ready!");
+    }
+
+    Console.newline(2);
+
+    config.parseBody();
+
+
+    // get a progress logger
+    ProgressLoggerWrapper progLog =
+        new ProgressLoggerWrapper(super.getProject(), config.getAssignmentName());
+
+    // start task submission
+    progLog.started();
+    initTasks(config.size(), this.getLogger());
+
+    if (totalTasks > 0) {
+      // submit commands to executor
+      for (Command cmd : config) {
+        // configure command
+        cmd.setCallback((Command.Callback) GatorGradleReportTask::completedTask);
```

Comment:
Because the report task depends on grading to be done, instead of duplicating code, we should figure out how to either make the report task `dependsOn` the grading task, and retrieve it's `CommandOutputSummary` (which does need to be static, so that may be fairly easy if things are executed in the same java instance), or create a grading task manually here and run it, instead of duplicating code, although this would still be kind of ugly.

Reply:
I'm familiar with the `dependsOn ` task, I will see if can fix it with this approach